packages/blockly-field-slider/src/field_slider.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Number slider input field.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * Class for an slider number field.
+ * @param {string|number=} opt_value The initial value of the field. Should cast
+ *    to a number. Defaults to 0.
+ * @param {?(string|number)=} opt_min Minimum value.
+ * @param {?(string|number)=} opt_max Maximum value.
+ * @param {?(string|number)=} opt_precision Precision for value.
+ * @param {?Function=} opt_validator A function that is called to validate
+ *    changes to the field's value. Takes in a number & returns a validated
+ *    number, or null to abort the change.
+ * @param {Object=} opt_config A map of options used to configure the field.
+ *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/number#creation}
+ *    for a list of properties this parameter supports.
+ * @extends {Blockly.FieldNumber}
+ * @constructor
+ */
+export const FieldSlider = function(opt_value, opt_min, opt_max, opt_precision,
```

Comment:
We have babel to transpile, curious why you don't use a class here?

Reply:
I'm going to look into that. I was following how most of the other blocks have been extended, but I would like to try to use a class, I just need to look up a few syntax things.

Reply:
Done.